controllers/discovery/initDiscoveryUserData.js

```diff
@@ -107,8 +108,9 @@ const InitDiscoveryUserData = async (req, res) => {
           ${where_anon_dm} 
           ${filterBlockedUser}
         ORDER BY
-          is_followed DESC,
-          COALESCE(A.karma_score, 0) DESC,
+          recently_active DESC,
+          community_info DESC,
+          A.karma_score DESC,
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The ordering of the results has been changed to primarily sort by the `recently_active` field and secondarily by the `community_info` field. This should work as expected as long as these fields are populated correctly. However, it's worth noting that if there are many users who have the same `recently_active` and `community_info` values, the order might not be consistent across different queries due to the lack of a unique secondary sorting field. Consider adding another unique field (like `user_id`) at the end of the ORDER BY clause to ensure consistent ordering.

```diff
- ORDER BY
-   recently_active DESC,
-   community_info DESC,
-   A.karma_score DESC,
-   followersCount DESC
+ ORDER BY
+   recently_active DESC,
+   community_info DESC,
+   A.karma_score DESC,
+   followersCount DESC,
+   A.user_id ASC
```
This change ensures that the result set is consistently ordered even when multiple users have the same `recently_active`, `community_info`, `karma_score`, and `followersCount` values.


<!-- This is an auto-generated comment by OSS CodeRabbit -->